### PR TITLE
New package: OpenChemistry.Avogadro2 version 1.99.0

### LINKED ISSUE #156209: [Package Request]: Avogadro 2
### How can we help?

I would like someone else to build the manifest.

### Please read and ensure the following

- [X] The installer meets the above requirements

### Please provide the following information

Download Page Url: https://www.openchemistry.org/downloads/
Publisher: [Open Chemistry project](https://www.openchemistry.org/about/)
Package Name: Avogadro 2
Description: 

```
Avogadro 2In order to tackle molecular simulation and visualization challenges in key areas of materials science, chemistry and biology it is necessary to move beyond fixed software applications. The Avogadro project is in the final stages of an ambitious rewrite of its core data structures, algorithms and visualization capabilities. The project began as a grass roots effort to address deficiencies observed by many of the early contributors in existing commercial and open source solutions. Avogadro is now a robust, flexible solution that can tie in to and harness the power of VTK for additional analysis and visualization capabilities.
Avogadro 2 is a chemical editor and visualization application, it is also a set of reusable software libraries written in C++ using principles of modularity for maximum reuse. The development of the first generation Avogadro application and library is documented in our paper, and this remains the preferred method of citation at present. The motivation for rewriting Avogadro, along with improvements and changes made in Avogadro 2 are summarized in our Source article. We provide a set of permissively licensed, open source, cross platform software components in the Avogadro 2 libraries, along with an end-user application with full source code, and binaries.
The library features updated and improved rendering, where we built upon the abstraction provided by previous API, but implemented a simple scene graph. This makes use of features such as impostor sphere rendering, resulting in significant rendering speed improvements while improving the quality of the visualization. The core is built for scalability, looking to enable the analysis of larger chemical structures and simulations being produced by computational chemistry codes today. Emphasis has also been placed on making it even easier to extend, using simple Python scripts to add simulation input capabilities, and data input/output along with access to full-blown C++ plugin APIs where more control is required.
Avogadro is now able to make full use of the visualization capabilities of VTK, in addition to its own powerful rendering capabilities. This means that complex visualization, involving techniques such as volume rendering for point data, or streamlines for vector fields, will now become possible. The project is composed of two separate repositories, with the `avogadroapp’ repository offering a full demonstration of how to use the libraries in an end-user application. The `avogadrolibs’ repository contains all of the libraries, with the option to only build subsets. The development process uses distributed version control (Git), testing (CTest/CDash), and automated binary generation.
```

Package Version: `1.91.0`
Installer URL: https://github.com/OpenChemistry/avogadroapp/releases/download/1.91.0/Avogadro2-1.91.0-win64.exe

### PR DESCRIPTION
**Pull request by Dragon1573**
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] This resolves #156209

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/156621)

### CODE REVIEW

**Comment by stephengillie** [comment]
URL: http://matebook-x-pro/d/programs/Avogadro2-1.99.0-win64.exe?sign=jOYBLNXHXjifcOaG13gPqQma6lo9ZFgKIUCMmDsdOmI=:0 
No such host is known. (matebook-x-pro:80)

(Automated message - build 883)

**Comment by stephengillie** [comment]
Automatic Validation ended with:
> URI: http://matebook-x-pro/d/programs/Avogadro2-1.99.0-win64.exe?sign=jOYBLNXHXjifcOaG13gPqQma6lo9ZFgKIUCMmDsdOmI=:0, Validation result: Failed, Error Message: No such host is known. (matebook-x-pro:80) No such host is known.

(Automated response - build 883.)

File: manifests/o/OpenChemistry/Avogadro2/1.99.0/OpenChemistry.Avogadro2.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+Scope: machine
+InstallModes:
+- interactive
+- silent
+- silentWithProgress
+UpgradeBehavior: install
+ReleaseDate: 2024-02-11
+Installers:
+- Architecture: x64
+  InstallerUrl: http://matebook-x-pro/d/programs/Avogadro2-1.99.0-win64.exe?sign=jOYBLNXHXjifcOaG13gPqQma6lo9ZFgKIUCMmDsdOmI=:0
```
Comment (stephengillie) [inline manifests/o/OpenChemistry/Avogadro2/1.99.0/OpenChemistry.Avogadro2.installer.yaml:16]: Hi @Dragon1573,

Is this URL local to your PC?

Comment (Dragon1573) [inline manifests/o/OpenChemistry/Avogadro2/1.99.0/OpenChemistry.Avogadro2.installer.yaml:16]:
```suggestion
  InstallerUrl: https://github.com/OpenChemistry/avogadroapp/releases/download/1.99.0/Avogadro2-1.99.0-win64.exe
```

Forget to switch back the official URL 🤣

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 
> Faulting application name: avogadro2.exe, version: 0.0.0.0, time stamp: 0x65c7a6e8 
> Faulting module name: unknown, version: 0.0.0.0, time stamp: 0x00000000 Exception code: 0xc000041d Fault offset: 0x0000000000000000 
> Faulting process id: 0x27bc 
> Faulting application start time: 0x01dab7bd6b930d1f 
> Faulting application path: C:\Program Files\Avogadro2\bin\avogadro2.exe 
> Faulting module path: unknown Report Id: 528bc2bc-335a-437c-a32b-f743d51ab1d6 
> Faulting package full name:  
> Faulting package-relative application ID:   Fault bucket 2238026438778600698, type 5 Event Name: BEX64 Response: Not available Cab Id: 0  Problem signature: P1: avogadro2.exe P2: 0.0.0.0 P3: 65c7a6e8 P4: StackHash_ac46 P5: 0.0.0.0 P6: 00000000 P7: PCH_DC_FROM_ntdll+0x000000000009D8C4 P8: c0000005 P9: 0000000000000008 P10:   Attached files: \\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5478.tmp.dmp \\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5506.tmp.WERInternalMetadata.xml \\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5507.tmp.xml \\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5522.tmp.csv \\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5533.tmp.txt  These files may be available here: \\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_avogadro2.exe_f8fcf29ed1384bcaa4b0d04652d5c8da46a5062_d7860816_66510817-4d4b-4383-afa2-40c83393393f  Analysis symbol:  Rechecking for solution: 0 Report Id: d2657093-bca1-4944-8a68-6a3951efc1c7 Report Status: 268435456 Hashed bucket: 9186cc54fb99b36bff0f11304154bcfa Cab Guid: 0

(Automated response - build 883.)

**Comment by Dragon1573** [comment]
Sorry, I can't get from manual validation logs. What can I try to resolve them?

**Comment by stephengillie** [comment]
These were in the Application log. The application didn't run, and didn't clearly say why. 

An Internet search brings back this issue being reported several times:
- https://discuss.avogadro.cc/t/avogadro-wont-start-on-windows-10/3077/2
  - version `v1.2.0n`
  - Dependency on `Microsoft.VCRedist.2010.x86`
- https://github.com/OpenChemistry/avogadrolibs/issues/1291
  - version `1.97.0`
  - Some students say the nightly builds work for them, so try one of those.
- https://github.com/OpenChemistry/avogadrolibs/issues/1431
  - version `1.98`
  - Closed due to driver bug and it's fixed in the nightly builds.
- https://discuss.avogadro.cc/t/cant-open-the-program-after-downloading/5287/
  - version `1.99.0`
  - Log files in `<USER>\AppData\Local\OpenChemistry\Avogadro\`
  - "Please try a nightly build if you are having problems with 1.99 on Windows."
  - "I am still having problem with nightly build one too. It gets installed but never runs or opens."

I'm not sure about the dependency, since it was for an earlier version, and wasn't suggested by the developers in the later links. What do you think?

**Comment by Dragon1573** [comment]
> These were in the Application log.

Sorry, I can't find them. Seems the application itself doesn't provide a shortcut to the log folder.

> The application didn't run, and didn't clearly say why.

There's nothing unusual for me. I'm not able to reproduce the problem.

image
image

**Comment by stephengillie** [comment]
I'm guessing that the reason it works for some and not others is still due to a poorly-docoumented dependency on a `VCRedist` or similar framework - just not the one I quoted above. 

Do you have any frameworks installed? Could one of them be a dependency for this application?

**Comment by Dragon1573** [comment]
> I'm guessing that the reason it works for some and not others is still due to a poorly-docoumented dependency on a `VCRedist` or similar framework - just not the one I quoted above. 
> 
> Do you have any frameworks installed? Could one of them be a dependency for this application? 

Nope...

I just copy these manifests I wrote "as is" and save it to my local device. Then I launch the test with `Tools/SandboxTest.ps1` script. In this case, I can't sure if there any pre-installed frameworks on my Sandbox instance.

For `Microsoft.VCRedist.*` dependencies, there're `vcruntime140.dll` bundled in the `bin/` folder of this package. I think it's not the point.

**Comment by stephengillie** [comment]
I do not know why it doesn't work on my VMs, but I'm willing to accept that there is something added to them, or missing from them, which is preventing the application from installing correctly. I sure wish someone knew what it was, so it could be added to this manifest.

**Review by stephengillie — APPROVED** [review]

**Pull request merged**